app.py

```diff
@@ -1,49 +1,113 @@
+    page_title="FinNetra",
 )
 
 
-def main():
-    st.markdown("<br>", unsafe_allow_html=True)
-    st.title("Starting app")
-    # audio = audiorecorder("Click to record", "Click to stop recording")
-    placeholder_story = """
-    Nama saya Budi Santoso, seorang pria berusia 35 tahun yang bekerja sebagai Manajer Pemasaran. Saya menikah dan memiliki seorang anak berusia 4 tahun. Setiap bulan, saya menerima gaji sebesar Rp 15.000.000 dan mendapatkan tambahan pendapatan Rp 2.000.000 dari bisnis sampingan. Saat ini, saya memiliki tabungan sebesar Rp 50.000.000 dan telah berinvestasi Rp 20.000.000 di reksa dana.
+@st.cache_resource
+def load_whisper_model():
+    """
+    Load and cache the medium size OpenAI Whisper model.
+    Returns:
+        object: The loaded Whisper model.
+    """
+    model = whisper.load_model("medium")
+    return model
+
+
+def transcribe_audio(model, audio):
+    """
+    Transcribe audio using the OpenAI Whisper model.
+    This function takes an audio object, ensures that the audio parameters
+    (frame rate, sample width, and channels) meet OpenAI Whisper specifications,
+    converts the audio data into a normalized numpy array, and then transcribes
+    it using the Whisper model.
+    Args:
+        model: The Whisper model used for audio transcription.
+        audio: The audio object to be transcribed.
+    Returns:
+        str: The transcribed audio result in text format.
+    """
+    if audio.frame_rate != 16000:
+        audio = audio.set_frame_rate(16000)
+    if audio.sample_width != 2:
+        audio = audio.set_sample_width(2)
+    if audio.channels != 1:
+        audio = audio.set_channels(1)
+
+    audio_array = np.array(audio.get_array_of_samples())
+    normalization_factor = 32768.0
+    audio_array = audio_array.astype(np.float32) / normalization_factor
 
-Namun, saya juga memiliki kewajiban finansial yang harus dipenuhi. Saya sedang mencicil Kredit Pemilikan Rumah (KPR) dengan sisa hutang sebesar Rp 500.000.000, di mana saya harus membayar cicilan bulanan Rp 5.000.000 selama 20 tahun ke depan. Selain itu, saya memiliki hutang kartu kredit sebesar Rp 10.000.000 dengan bunga 2% per bulan.
+    options = dict(
+        language=os.environ["WHISPER_MODEL_LANGUAGE"], beam_size=5, best_of=5
+    )
 
-Pengeluaran bulanan saya terdiri dari cicilan KPR Rp 5.000.000, cicilan kartu kredit Rp 500.000, biaya hidup sehari-hari seperti kebutuhan rumah tangga, sekolah anak, dan transportasi sebesar Rp 8.000.000, serta pengeluaran lain-lain sebesar Rp 2.000.000. Saya juga menyisihkan Rp 1.000.000 setiap bulan untuk tabungan dan investasi. Total pengeluaran bulanan saya mencapai Rp 16.500.000, sehingga saya memiliki sisa pendapatan bulanan Rp 500.000
+    transcribe_options = dict(task="transcribe", **options)
+    with st.spinner("Transcribing Audio"):
+        result = model.transcribe(audio_array, **transcribe_options)
+    return result["text"]
+
+
+def speak(text: str):
+    """
```

Comment:
Try to explore the possibility whether or not we need to pass `pre_processor_funcs`, try to compare the results and see the tradeoff e.g. speed vs result (does it speak better but with higher response time, does it speak worse, etc.)